src/udev/net/link-config.c

```diff
@@ -834,6 +867,14 @@ int config_parse_wol_password(
         assert(userdata);
 
         if (isempty(rvalue)) {
+                link->wol_password_from_cred = false;
+                link->wol_password = erase_and_free(link->wol_password);
+                return 0;
+        }
+
+        r = parse_boolean(rvalue);
```

Comment:
hmm, this is definitely a workable approach, but i wonder if it wouldn't be simpler and more natural to simply say that if `WakeOnLan=secureon` is selected but `WakeOnLanPassword=` is not we'll search for the password in the creds? And then not have any boolean parsing here at all?

That way: 

`WakeOnLan=secureon` without `WakeOnLanPassword=` → search for the password in the creds and use it if it exists
`WakeOnLanPassword=` without `WakeOnLan=` → imply `WakeOnLan=secureon`